RecoLocalCalo/HGCalRecProducers/test/HGCalLocalRecoTask.cc

```diff
@@ -0,0 +1,252 @@
+  HGCHEFdigiCollection_       = ps.getParameter<std::string>("HGCHEFdigiCollection");
+  HGCHEBdigiCollection_       = ps.getParameter<std::string>("HGCHEBdigiCollection");
+
+  
+  if ( outputFile_.size() != 0 ) {
+    edm::LogInfo("HGCalLocalRecoTaskInfo") << "histograms will be saved to '" << outputFile_.c_str() << "'";
+  } else {
+    edm::LogInfo("HGCalLocalRecoTaskInfo") << "histograms will NOT be saved";
+  }
+  
+  // verbosity switch
+  verbose_ = ps.getUntrackedParameter<bool>("verbose", false);
+ 
+  if ( verbose_ ) {
+    edm::LogInfo("HGCalLocalRecoTaskInfo") << "verbose switch is ON"; 
+  } else {
+    edm::LogInfo("HGCalLocalRecoTaskInfo") << "verbose switch is OFF";
+  }
+                                                                                                                                          
+  dbe_ = 0;
+                                                                                                                                          
+  // get hold of back-end interface
+  dbe_ = edm::Service<DQMStore>().operator->();
+                                                                                                                                          
+  if ( dbe_ ) {
+    if ( verbose_ ) {
+      dbe_->setVerbose(1);
+    } else {
+      dbe_->setVerbose(0);
+    }
+  }
+                                                                                                                                          
+  if ( dbe_ ) {
+    if ( verbose_ ) dbe_->showDirStructure();
+  }
+
+
+  meHGCEEUncalibRecHitOccupancy_ = 0;
+  meHGCHEFUncalibRecHitOccupancy_ = 0;
+  meHGCHEBUncalibRecHitOccupancy_ = 0;
+
+  Char_t histo[70];
+ 
+  if ( dbe_ ) {
+    dbe_->setCurrentFolder("HGCalLocalRecoTask");
+  
+    sprintf (histo, "HGCalLocalRecoTask HGCEE occupancy" ) ;
+    meHGCEEUncalibRecHitOccupancy_ = dbe_->book2D(histo, histo, 360, 0., 360., 170, -85., 85.);
+
+    sprintf (histo, "HGCalLocalRecoTask HGCHEF occupancy" ) ;
+    meHGCHEFUncalibRecHitOccupancy_ = dbe_->book2D(histo, histo, 360, 0., 360., 170, -85., 85.);
+
+    sprintf (histo, "HGCalLocalRecoTask HGCHEB occupancy" ) ;
+    meHGCHEBUncalibRecHitOccupancy_ = dbe_->book2D(histo, histo, 360, 0., 360., 170, -85., 85.);
+    
+  }
+ 
+}
+
+HGCalLocalRecoTask::~HGCalLocalRecoTask(){
+ 
+  if ( outputFile_.size() != 0 && dbe_ ) dbe_->save(outputFile_);
+
+}
+
+void HGCalLocalRecoTask::beginJob(){
+
+}
+
+void HGCalLocalRecoTask::endJob(){
+
+}
+
+void HGCalLocalRecoTask::analyze(const edm::Event& e, const edm::EventSetup& c)
+{
+
+  edm::LogInfo("HGCalLocalRecoTaskInfo") << " Run = " << e.id().run() << " Event = " << e.id().event();
+  
+   edm::Handle< HGCEEDigiCollection > pHGCEEDigis;
+   edm::Handle< HGCHEDigiCollection > pHGCHEFDigis;
+   edm::Handle< HGCHEDigiCollection > pHGCHEBDigis;
+
+   try {
+     //     e.getByLabel( digiProducer_, HGCEEdigiCollection_, pHGCEEDigis);
+     e.getByLabel( digiProducer_, pHGCEEDigis);
```

Comment:
I think that the try-catch is not needed here: just check the pHGCEEDigis.isValid()